plugins/module_utils/scale.py

```diff
@@ -33,6 +33,20 @@
 
 
 SCALE_ARG_SPEC = {
+    'resource_definition': {
+        'type': list_dict_str,
+        'aliases': ['definition', 'inline']
+    },
+    'src': {
+        'type': 'path',
+    },
+    'kind': {},
```

Comment:
would it make sense to group together the common parameters into logical pieces that various argspecs could make use of (and each group would have a corresponding doc fragement)? ie, `api_version`, `kind`, `name` and `namespace`, `resource_definition` and `src`, `wait`, `wait_timeout` and `wait_condition`. They wouldn't be inheriting argspecs from another module, just composing them from sets of shared parameters that ideally can just be updated in one place

Reply:
basically the same as we do with `AUTH_ARG_SPEC`